utils/compression.sql

```diff
@@ -0,0 +1,42 @@
+-- Update metadata for chunks compressed before 2.0.
+--
+-- After you have updated from pre-2.0 it might be that some
+-- compressed chunks are missing information about the approximate row
+-- count. To correct that, call this function. Note that it will only
+-- update approximate row counts that are NULL, so running it several
+-- times will not overwrite existing values.
+--
+-- To avoid blocking too many tables, it will commit after each chunk,
+-- which means that the function might be slow if there is a lot of
+-- data to update, but it will not block other queries.
+CREATE PROCEDURE correct_compression_chunk_size ()
+LANGUAGE PLPGSQL
+AS $$
+DECLARE
+  plain_chunk RECORD;
+  comp_chunk TEXT;
+  rowcount_pre BIGINT;
+  rowcount_post BIGINT;
+BEGIN
+  FOR plain_chunk IN
+      SELECT * FROM _timescaledb_catalog.chunk WHERE compressed_chunk_id IS NOT NULL
+  LOOP
+      SELECT format('%I.%I', schema_name, table_name) INTO comp_chunk
+      FROM _timescaledb_catalog.chunk
+      WHERE id = plain_chunk.compressed_chunk_id;
+      
+      EXECUTE format('SELECT sum(_ts_meta_count), count(*) FROM %s', comp_chunk) INTO rowcount_pre, rowcount_post;
+
+      -- We update them separately since either can be null (in
+      -- theory) and we do not want to overwrite non-null values.
+      UPDATE _timescaledb_catalog.compression_chunk_size
+         SET numrows_post_compression = rowcount_post
+       WHERE chunk_id = plain_chunk.id AND numrows_post_compression IS NULL;
```

Comment:
This can be done in a single update statement I think with a coalesce statement in the set clause, though I don't know why you don't want to overwrite non-null values, it seems like if one is null both should be null and the answer you have here should be accurate in either case, so there's no reason not to overwrite.

Reply:
I was mostly worried about updating values that does not need to be updated since they could have been set for some other reason (custom tweaks to hack around a problem, for example), but if you don't think that is an issue, I'm fine. Since we have an accurate row count anyway we can just as well update it for the rare case where one is null but not the other.

Reply:
Yeah. I think that's fine.